Dangerfile

```diff
@@ -0,0 +1,25 @@
+# Make sure code changes come with corresponding tests
+has_app_changes = !git.modified_files.grep(/lib/).empty?
+has_spec_changes = !git.modified_files.grep(/spec/).empty?
+
+if has_app_changes && !has_spec_changes
+  warn('There are code changes, but no corresponding tests. '\
+         'Please include tests if this PR introduces any modifications in '\
+         'behavior.',
+       sticky: false)
+end
+
+# Make it more obvious that a PR is a work in progress and shouldn't be merged yet
+warn('PR is classed as Work in Progress') if github.pr_title.include? '[WIP]'
+
```

Comment:
Might want to expand this a bit. I have seen Matt use "[DON'T MERGE]" (which implies WIP), and I have used "[WIP -- Don't merge]" which doesn't match the spec.

Something like:

```ruby
[/WIP/, /don\'t merge/i, /work in progress/i ].each do |wip|
  (warn('PR is classed as Work in Progress'); break) if github.pr_title =~ wip
end
```